yt_dlp/extractor/tudou.py

```diff
@@ -0,0 +1,80 @@
+from .common import InfoExtractor
+import time
+import re
+
+
+class TudouIE(InfoExtractor):
+    _VALID_URL = r'https?://(?:play\.)?tudou\.com/v_show/(?P<id>id_[\w=]+).html[\w\W]+'
+    _TESTS = [{
+        'url': 'https://play.tudou.com/v_show/id_XNjAxNjI2OTU3Ng.html?',
+        'md5': 'to be updated',
+        'info_dict': {
+            'id': 'XNjAxNjI2OTU3Ng==',
+            'ext': 'mp4',
+            'title': '阿斯塔意识到哈里杀了人，自己被骗了',
+            'show_name': '外星居民 第一季',
+        },
+        # 'skip': 'testing skip function',
+    }]
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        webpage = self._download_webpage(url, video_id)
+        data = self._search_json(r'window\.__INITIAL_DATA__\s*=', webpage, 'initial data', video_id)
+
+        # The .json file is much cleaner, thanks for the guide
+        video_id = data['data']['data']['data']['extra']['videoId']
+        title = data['data']['data']['data']['extra']['videoTitle']
+        show_name = data['data']['data']['data']['extra']['showName']
+
+        _, urlh = self._download_webpage_handle(
+            "https://log.mmstat.com/yt.gif", video_id, 'Retrieving cna info'
+        )
+        re_cna = re.compile(r'cna=([\w\W]+; expires)')
+        cna = re.findall(re_cna, urlh.headers['set-cookie'])[0].replace("; expires", "")
+        ts1 = time.time() * 1000
+
+        params1 = {
+            'vid': video_id,
+            'ccode': '050F',
+            'client_ip': '192.168.1.1',
+            'utid': cna,
+            'client_ts': ts1,
```

Comment:
This code all looks very similar to `YoukuIE`. Perhaps there needs to be a `YoukuBaseIE` from which this extractor and `YoukuIE` are subclassed